packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -47,16 +34,27 @@ export class Wallet {
   ) {}
 
   /**
-   * TODO: This is a basic implementation of createChannels.
-   * This will be cleaned up in https://github.com/statechannels/statechannels/issues/3365
-   * @param args
-   * @param numberOfChannels
-   * @returns
+   Creates channels using the given parameters.
+   * @param channelParameters The parameters to use for channel creation. A channel will be created for each entry in the array.
+   * @returns A promise that resolves to a collection of ObjectiveResult.
    */
-  public async createChannels(args: CreateChannelParams, numberOfChannels: number): Promise<void> {
-    const createResult = await this._engine.createChannels(args, numberOfChannels);
-
-    await this.ensureObjectives(createResult.newObjectives, getMessages(createResult.outbox));
+  public async createChannels(
+    channelParameters: CreateChannelParams[]
+  ): Promise<ObjectiveResult[]> {
+    return Promise.all(
+      channelParameters.map(async p => {
+        const createResult = await this._engine.createChannel(p);
+
+        const {newObjective, channelResult} = createResult;
+
+        return {
+          channelId: channelResult.channelId,
+          currentStatus: newObjective.status,
+          objectiveId: newObjective.objectiveId,
+          done: this.ensureObjective(newObjective, getMessages(createResult)),
```

Comment:
If `done` is never meant to reject, should we 
```suggestion
          done: this.ensureObjective(newObjective, getMessages(createResult)).catch(parseErrorSomehow?),
```

Reply:
That's a good point. I meant to wire that up to the `InternalError` type, so that any unhandled error gets caught and we just return an `InternalError` to the user.

Reply:
With the changes in 68a57ff I think we no longer need this, since we now are manually resolving the promises.